src/chat/components/FileAttachmentModal.tsx

```diff
@@ -0,0 +1,264 @@
+import React, {
+  forwardRef,
+  useCallback,
+  useImperativeHandle,
+  useState,
+} from 'react';
+import {
+  View,
+  TouchableOpacity,
+  Modal,
+  Text,
+  StyleSheet,
+  ScrollView,
+  Image,
+  ViewStyle,
+  StyleProp,
+  ImageStyle,
+  Pressable,
+  Alert,
+  Platform,
+  PermissionsAndroid,
+} from 'react-native';
+import DocumentPicker, {
+  type DocumentPickerResponse,
+} from 'react-native-document-picker';
+import {
+  type MediaType,
+  type IUserInfo,
+  type MessageProps,
+} from '../../interfaces';
+import uuid from 'react-native-uuid';
+import { formatSize, getAbsoluteFilePathWithName } from '../../utilities';
+
+interface FileAttachmentModalProps {
+  isVisible?: boolean;
+  buttonTitle?: string;
+  shareIcon?: number;
+  shareIconStyle?: StyleProp<ImageStyle>;
+  modalTitle?: string;
+  modalDescription?: string;
+  modalTitleStyle?: StyleProp<ViewStyle>;
+  modalDescriptionStyle?: StyleProp<ViewStyle>;
+  modalButtonStyle?: StyleProp<ViewStyle>;
+  modalButtonTextStyle?: StyleProp<ViewStyle>;
+  onSend: (message: MessageProps) => void;
+  userInfo?: IUserInfo | null;
+}
+
+export interface FileAttachmentModalRef {
+  show: () => void;
+  hide: () => void;
+}
+
+const FileAttachmentModal = forwardRef<
+  FileAttachmentModalRef,
+  FileAttachmentModalProps
+>((props, ref) => {
+  const {
+    buttonTitle = 'Choose files',
```

Comment:
The modal title is 'Share a file', so the button should be "Chose a file" or "Select file" right?

Reply:
this modal only show to user can choose their files on devices. I used same text with Messenger button on facebook before

Reply:
I see on Messenger, It able to choose multiple files, but here you can choose one file at a moment